### PR TITLE
TD-10, add POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by marvinobig**
## This Ticket Includes:
- Tests for 200 status on path "/api/articles/:article_id/comments"
- Tests for response object on path "/api/articles/:article_id/comments"
- Tests for 400 status response on path "/api/articles/:article_id/comments"
- Improvement of error handling for psql errors

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -248,6 +249,66 @@ describe("GET /api/articles/:article_id/comments", () => {
...
 
+describe("POST /api/articles/:article_id/comments", () => {
+  test("status:200, should respond with a status of 200", () => {
+    const newComment = { username: "rogersop", body: "this rocks" };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(newComment)
+      .expect(200);
+  });
+  test("status:200, should respond with the object that was sent", () => {
```
Comment (k-huggs) [inline __tests__/app.test.js:260]: you're not testing for a status 200 hundred with this test, combine with test above

Comment (k-huggs) [reply __tests__/app.test.js:260]: Also if you're posting think about what status code you would want

File: __tests__/app.test.js [hunk 2]
```diff
@@ -248,6 +249,66 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    const newComment = { username: "", body: "" };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Request Body Includes Incorrect Data");
+      });
+  });
+  test("status: 400, should respond with an error message when article does not exist", () => {
```
Comment (k-huggs) [inline __tests__/app.test.js:300]: this would be a 404 not a 400

File: controllers/api.controller.js [hunk 3]
```diff
@@ -26,30 +31,47 @@ exports.updateArticleByIdController = async (req, res, next) => {
...
+
+exports.postArticleCommentByIdController = async (req, res, next) => {
+  try {
+    const commentData = req.body;
+    const id = Number(req.params.article_id);
+    const newComment = await insertArticleCommentById(id, commentData);
+    const newCommentObj = { newComment };
 
-    res.status(200).send(articleComments);
+    res.status(200).send(newCommentObj);
```
Comment (k-huggs) [inline controllers/api.controller.js:74]: status code for post

File: __tests__/app.test.js [hunk 4]
```diff
@@ -248,6 +249,66 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    const newComment = { username: "rogersop", body: "this rocks" };
+    return request(app)
+      .post("/api/articles/100/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Request Body Includes Incorrect Data");
+      });
+  });
```
Comment (k-huggs) [inline __tests__/app.test.js:309]: could add a test for if username does not exist

**Comment by k-huggs** [comment]
check the comments, some slight amendments to be made

File: __tests__/app.test.js (hunk 4 above)
Comment (marvinobig) [reply __tests__/app.test.js:303]: I've tried that, but it returns the same error code "23503" regardless so I put it under the same error

File: __tests__/app.test.js [hunk 5]
```diff
@@ -248,6 +249,64 @@ describe("GET /api/articles/:article_id/comments", () => {
...
 
+describe("POST /api/articles/:article_id/comments", () => {
+  test("status:201, should respond with the object that was sent", () => {
+    const newComment = { username: "rogersop", body: "this rocks" };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body: { newComment } }) => {
+        expect(newComment[0]).toBeInstanceOf(Object);
```
Comment (alexdevdavis) [inline __tests__/app.test.js:260]: you're expecting an object back but are receiving an array (I'd suggest extracting the object from its array in the model

File: __tests__/app.test.js [hunk 6]
```diff
@@ -248,6 +249,64 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+  });
+  test("status:404, should respond with error message when given username that does not exist", () => {
+    const newComment = { username: 444, body: "" };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe(
+          'Key (author)=(444) is not present in table "users".'
```
Comment (alexdevdavis) [inline __tests__/app.test.js:292]: avoid revealing specific table / column names in your db

File: __tests__/app.test.js [hunk 7]
```diff
@@ -248,6 +249,64 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+  });
+  test("status:404, should respond with an error message when given article does not exist", () => {
+    const newComment = { username: "rogerso", body: "this rocks" };
+    return request(app)
+      .post("/api/articles/100/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe(
+          'Key (article_id)=(100) is not present in table "articles".'
```
Comment (alexdevdavis) [inline __tests__/app.test.js:304]: protect your db (as per above comment)

**Review by alexdevdavis — APPROVED** [review]
nicely done

**Pull request merged**